ha/k8s_setup/ha_start.py

```diff
@@ -37,24 +37,50 @@ def handle_signal(signum, frame):
     Signal handler call beck function, to forward the received signal to driver process
     """
     sys.stdout.write(f"Signal {signum} at frame {frame} received.")
-    sys.stdout.write(f"Sending {signum} to driver process pid: {driver_process.pid}.")
-    driver_process.send_signal(signum)
-
-try:
-    if args.services not in service_entry_mapping:
-        sys.stdout.write('Please provide a valid service name')
-        usage('ha_start')
-        sys.exit(22)
-    # setting Signal handlers for all standered signals
+    # If received SIGUSR1 then restart service
+    if signal.SIGUSR1.value == signum:
+        print("\nCaught signal.SIGUSR1, Restarting process")
+        driver_process.kill()
+        set_signals()
+        start_process()
+        driver_process.wait()
+    else:
+        sys.stdout.write(f"Sending {signum} to driver process pid: {driver_process.pid}.")
+        driver_process.kill()
```

Comment:
please use this 'driver_process.send_signal(signum)' not kill(), as I am forwarding all the slandered signals not only sigterm.